src/overlay/FlowControlCapacity.cpp

```diff
@@ -75,8 +75,55 @@ FlowControlByteCapacity::getCapacityLimits() const
 uint64_t
 FlowControlByteCapacity::getMsgResourceCount(StellarMessage const& msg) const
 {
-
-    return static_cast<uint64_t>(xdr::xdr_argpack_size(msg));
+    switch (msg.type())
```

Comment:
How is this different from the generated implementation?

Reply:
we need to exclude size of `type` (additional 4 bytes) for the calculation to be consistent with the rest of the codebase. I updated this function to be less verbose though.